tools/astarte_export/lib/astarte/fetchdata/fetchdata.ex

```diff
@@ -285,9 +285,42 @@ defmodule Astarte.Export.FetchData do
     end
   end
 
+  defp from_native_type(value, :binaryblobarray) when is_list(value) do
+    case value do
+      list when is_list(list) -> Enum.map(list, fn x -> from_native_type(x, :binaryblob) end)
+      _ -> value
+    end
+  end
+
   defp from_native_type(value, :binaryblob) do
-    {:ok, binary_blob} = Base.encode64(value)
-    binary_blob
+    case value do
+      <<_::binary>> -> Base.encode64(value)
+      _ -> value
+    end
+  end
+
+  defp from_native_type(value, :booleanarray) do
+    Enum.map(value, fn element -> to_string(element) end)
+  end
+
+  defp from_native_type(value, :datetimearray) do
+    Enum.map(value, fn element -> to_string(element) end)
+  end
+
+  defp from_native_type(value, :doublearray) do
+    Enum.map(value, fn element -> to_string(element) end)
+  end
+
+  defp from_native_type(value, :integerarray) do
+    Enum.map(value, fn element -> to_string(element) end)
+  end
+
+  defp from_native_type(value, :longintegerarray) do
+    Enum.map(value, fn element -> to_string(element) end)
+  end
+
+  defp from_native_type(value, :stringarray) do
+    Enum.map(value, fn element -> to_string(element) end)
```

Comment:
Rather than rewriting the same function for all array types (except for `:binaryarray` and `:datetimearray`) you could define a list of accepted atoms (e.g. `~w(booleanarray datetimearray doublearray integerarray longintegerarray stringarray)a`) and just check in the guard:
```elixir
defp from_native_type(value, native_type) when native_type in @native_types do
  Enum.map(value, &to_string/1)
end
```

Reply:
Thank you for the suggestion, it's an excellent approach to improve code efficiency and clarity.